system/HTTP/CURLRequest.php

```diff
@@ -309,7 +309,11 @@ protected function parseOptions(array $options)
 
         if (array_key_exists('headers', $options) && is_array($options['headers'])) {
             foreach ($options['headers'] as $name => $value) {
-                $this->setHeader($name, $value);
+                if (is_scalar($value)) {
+                    $this->setHeader($name, $value);
```

Comment:
Is there any reason to include stringables?
```suggestion
                if (is_scalar($value) || (is_object($value) && method_exists($value, '__toString')) {
                    $this->setHeader($name, (string) $value);
```